src/system.rs

```diff
@@ -329,6 +416,7 @@ impl Resolver {
         &mut self,
         id: A,
         type_store: &mut TypeStore,
+        load_isl_import: Option<&IslImport>, // if its the root of schema set it to None otherwise will be set to the IslImport to be loaded
```

Comment:
Sorry I should have been more clear on this. I meant:
```rust
// If we are loading the root schema then this will be set to `None` ( i.e. in the beginning when 
// this method is called from the load_schema method of schema_system it is set to `None`)
// Otherwise if we are loading an import of the schema then this will be set to `Some(isl_import)` 
// to be loaded (i.e. Inside schema_from_elements while loading imports this will be set to 
// `Some(isl_import)`)
```
Does this make sense? I can add more clarification in the comments.

Reply:
Yes, that makes sense.